### PR TITLE
SYN-6727: more context for easy perm auth deny

### PR DESCRIPTION
**Pull request by MichaelSquires**
feat: Added context info to the easyperm struct for reporting in AuthDeny.
feat: Added info argument to easyperms init functions.

### CODE REVIEW

File: synapse/lib/stormlib/easyperm.py [hunk 1]
```diff
@@ -2,6 +2,7 @@
 import synapse.common as s_common
 
 import synapse.lib.cell as s_cell
+import synapse.lib.schemas as s_schemas
```
Comment (MichaelSquires) [inline synapse/lib/stormlib/easyperm.py:5]: remove me

File: synapse/tests/test_lib_stormlib_easyperm.py [hunk 2]
```diff
@@ -91,3 +96,29 @@ async def test_stormlib_easyperm_basics(self):
 
             with self.raises(s_exc.BadArg):
                 await core.callStorm('$lib.auth.easyperm.init(({}), default=(6))')
+
+    async def test_stormlib_easyperm_auth_info(self):
+
+        async with self.getTestCore() as core:
+
+            visi = await core.auth.addUser('visi')
```
Comment (MichaelSquires) [inline synapse/tests/test_lib_stormlib_easyperm.py:104]: remove me

File: synapse/tests/test_lib_stormlib_easyperm.py [hunk 3]
```diff
@@ -91,3 +96,29 @@ async def test_stormlib_easyperm_basics(self):
...
+            someuser = await core.auth.addUser('someuser')
+
+            #$lib.auth.easyperm.confirm($a, $lib.auth.easyperm.level.admin)
```
Comment (MichaelSquires) [inline synapse/tests/test_lib_stormlib_easyperm.py:107]: remove me

**Review by invisig0th — CHANGES_REQUESTED** [review]

File: synapse/lib/cell.py [hunk 4]
```diff
@@ -3028,18 +3028,21 @@ def _reqEasyPerm(self, item, user, level, mesg=None):
...
+        info = item['permissions']['info']
+
         if mesg is None:
+            infomsg = ' '.join([f'{key}={valu}' for key, valu in info.items()])
+            if infomsg:
+                infomsg = ' ' + infomsg
+
             permname = permnames.get(level)
-            mesg = f'User ({user.name}) has insufficient permissions (requires: {permname}).'
+            mesg = f'User ({user.name}) has insufficient easyperms level (requires: {permname}){infomsg}.'
```
[the remark this answers is not in the document]
Comment (MichaelSquires) [reply synapse/lib/cell.py:3043]: I was trying to distinguish between easyperms and regular permissions so the users would (hopefully) understand that they can't use the optic permissions editor to fix this. This was a suggestion from @mikemoritz who submitted the ticket on behalf of a customer.

File: synapse/lib/schemas.py [hunk 5]
```diff
@@ -14,9 +18,24 @@
...
         },
+        'default': {
+            'type': 'integer',
+            'enum': [PERM_DENY, PERM_READ, PERM_EDIT, PERM_ADMIN],
+        },
+        'info': {
+            'type': 'object',
+            'default': {},
+            'properties': {
+                'type': {'type': 'string', 'minLen': 2},
```
[the remark this answers is not in the document]
Comment (MichaelSquires) [reply synapse/lib/schemas.py:29]: Yeah, maybe? My intent with `info` was to have data that was useful for reporting errors only. The other option is to set `additionalProperties: true` on `info` and just let the keys/values be freeform.

[the remark this answers is not in the document]
Comment (MichaelSquires) [reply synapse/lib/schemas.py:29]: Maybe it should be name `errinfo` or something more specific?

File: synapse/tests/test_lib_stormlib_vault.py [hunk 6]
```diff
@@ -76,7 +76,7 @@ async def test_stormlib_vault(self):
 
             opts = {'vars': {'iden': uiden}, 'user': visi2.iden}
             msgs = await core.stormlist('$vault = $lib.vault.get($iden)', opts=opts)
-            self.stormIsInErr(f'User (visi2) has insufficient permissions (requires: read).', msgs)
+            self.stormIsInErr(f'User (visi2) has insufficient easyperms level (requires: read) type=vault iden={uiden} cmd=vault.set.perm', msgs)
```
[the remark this answers is not in the document]
Comment (MichaelSquires) [reply synapse/tests/test_lib_stormlib_vault.py:79]: This will get updated accordingly when we settle on the exception message.

**Pull request closed without merging**